### PR TITLE
Add base 64 encoding support

### PR DESCRIPTION
**Pull request by marschall**

### CODE REVIEW

**Comment by jbrichau** [comment]
To merge this, we need an implementation on all tested platforms.

**Comment by marschall** [comment]
> To merge this, we need an implementation on all tested platforms.

Alright, I'll see if I can do something.

**Comment by jbrichau** [comment]
@marschall can I write to your fork? I guess it will be easier if I take a stab at the GemStone implementations?

**Comment by marschall** [comment]
> @marschall can I write to your fork? I guess it will be easier if I take a stab at the GemStone implementations?

Sure, please go ahead.

**Comment by marschall** [comment]
I just noted Gemstone 3.5 added support for Base64

https://downloads.gemtalksystems.com/docs/GemStone64/3.5.x/GS64-ReleaseNotes-3.5/3-ProgrammingChanges.htm#pgfId-1968321

I wonder how much back we want to support, same for Squeak and Pharo.

**Comment by jbrichau** [comment]
> I just noted Gemstone 3.5 added support for Base64

Actually, Base64 encoding was part of the GLASS project since a long time. See https://github.com/glassdb/glass/tree/044b42d4601d5ca3789ade45c7a3e6bcdca9c7f9/repository/Squeak.package/Base64MimeConverter.class

> 
> https://downloads.gemtalksystems.com/docs/GemStone64/3.5.x/GS64-ReleaseNotes-3.5/3-ProgrammingChanges.htm#pgfId-1968321
> 
> I wonder how much back we want to support, same for Squeak and Pharo.

I just recently dropped Pharo 6 but we could start dropping Pharo 7 as well. I think Pharo 8 is best kept working for a little while longer. Most people are on Pharo 9 or even Pharo 10.

The Gemstone versions are probably soon going to be cut off since anything older than 3.5 is not running anymore on ubuntu 18, which is not going to be running on github actions anymore on dec 1st...  I did not count on that sooner than the real standard support enddate (april 2023) but ok...

File: repository/Grease-GemStone-Core.package/GRGemStonePlatform.class/instance/base64Encode..st [hunk 1]
```diff
@@ -0,0 +1,3 @@
+encoding
+base64Encode: aByteArray
+	^ (Base64MimeConverter base64Encode: aByteArray readStream) contents
```
Comment (jbrichau) [inline repository/Grease-GemStone-Core.package/GRGemStonePlatform.class/instance/base64Encode..st]: replace `base64Encode:` with `mimeEncode:` ;-)

Comment (jbrichau) [reply repository/Grease-GemStone-Core.package/GRGemStonePlatform.class/instance/base64Encode..st]: Quickly did that...

**Review by jbrichau — APPROVED** [review]

**Pull request merged**